x/commitment/types/commitments.go

```diff
@@ -101,6 +101,9 @@ func (c *Commitments) DeductFromCommitted(denom string, amount math.Int, currTim
 			if lockedAmount.GT(c.CommittedTokens[i].Amount) {
 				return errors.Wrapf(ErrInsufficientWithdrawableTokens, "amount: %s denom: %s", amount, denom)
 			}
+			if c.CommittedTokens[i].Amount.IsZero() {
+				c.CommittedTokens = append(c.CommittedTokens[:i], c.CommittedTokens[i+1:]...)
+			}
```

Comment:
@cryptokage1996 can you also add a migration that will remove the existing commitments with a zero amount, so we can get rid of them in Testnet, thanks